src/Microsoft.Health.SqlServer/ManagedIdentitySqlConnection.cs

```diff
@@ -0,0 +1,53 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Services.AppAuthentication;
+using Microsoft.Data.SqlClient;
+using Microsoft.Health.SqlServer.Configs;
+
+namespace Microsoft.Health.SqlServer
+{
+    public class ManagedIdentitySqlConnection : ISqlConnection
+    {
+        private readonly SqlServerDataStoreConfiguration _sqlServerDataStoreConfiguration;
+
+        public ManagedIdentitySqlConnection(SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration)
+        {
+            EnsureArg.IsNotNull(sqlServerDataStoreConfiguration, nameof(sqlServerDataStoreConfiguration));
+
+            _sqlServerDataStoreConfiguration = sqlServerDataStoreConfiguration;
+        }
+
+        /// <summary>
+        /// Get sql connection after getting access token.
+        /// </summary>
+        /// <param name="connectToMaster">Should connect to master?</param>
+        /// <returns>Sql connection task.</returns>
+        public async Task<SqlConnection> GetSqlConnectionAsync(bool connectToMaster = false)
+        {
+            EnsureArg.IsNotNullOrEmpty(_sqlServerDataStoreConfiguration.ConnectionString);
+
+            SqlConnectionStringBuilder connectionBuilder = connectToMaster ?
+                new SqlConnectionStringBuilder(_sqlServerDataStoreConfiguration.ConnectionString) { InitialCatalog = string.Empty } :
+                new SqlConnectionStringBuilder(_sqlServerDataStoreConfiguration.ConnectionString);
+
+            SqlConnection sqlConnection = new SqlConnection(connectionBuilder.ToString());
+
+            AzureServiceTokenProvider azureServiceTokenProvider = new Microsoft.Azure.Services.AppAuthentication.AzureServiceTokenProvider();
```

Comment:
We should not be creating new instance of AzureServiceTokenProvider every time. This will result call to AAD to get new token every time a connection is created.

Reply:
Apologies, I missed to make these changes. I will make _azureServiceTokenProvider member of the class and access that. Since ManagedIdentitySqlConnection is registered as singleton service, _azureServiceTokenProvider should not be created every time. Please correct me if this understanding is not right.